src/py21cmmc/likelihood.py

```diff
@@ -965,3 +965,122 @@ def define_noise(self, ctx, model):
             return [{"sigma": s(m["Muv"])} for s, m in zip(sig, model)]
         else:
             return [{"sigma": s} for s in sig]
+
+
+class LikelihoodEDGES(LikelihoodBaseFile):
+    """
+    A likelihood based on chi^2 comparison to Global Signal of EDGES timing and FWHM
+
+    This is the likelihood arising from Bowman et al. (2018), which reports an absorption feature
+    in the 21-cm brightness temperature spectra
+
+    Parameters
+    ----------
+
+    use_width : bool
+        whether to use the FWHM in the likelihood, by default it's False
+
+    """
+
+    FREQ_EDGES = 78.
+    FREQ_ERR_EDGES = 1.
+    FWHM_EDGES = 19.
+    FWHM_ERR_UPP_EDGES = 4.
+    FWHM_ERR_LOW_EDGES = 2.
+
+    required_cores = [core.CoreLightConeModule]
+
+    def __init__(self, use_width=False, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.use_width = use_width
+
+    def reduce_data(self, ctx):
+        frequencies=1420.0 / (np.array(ctx.get("lightcone").node_redshifts) + 1)
+        global_signal=ctx.get("lightcone").global_brightness_temp
+        global_signal_interp = InterpolatedUnivariateSpline(frequencies, global_signal,k=4)
+        cr_pts = global_signal_interp.derivative().roots()
+        cr_vals = global_signal_interp(cr_pts)
+        if (len(cr_vals) == 0):
+            return dict(Freq_Tbmin = None, FWHM = None) # there is no solution -- global signal never reaches a minimum
+        else:
+            Freq_Tbmin = cr_pts[np.argmin(cr_vals)]
+            if not self.use_width:
+                return dict(Freq_Tbmin = Freq_Tbmin,  FWHM = None)
+            # calculating the frequencies when global signal = the FWHM
+            Freqs_HM= InterpolatedUnivariateSpline(frequencies,global_signal-global_signal_interp(Freq_Tbmin) * 0.5, k=3).roots()
```

Comment:
again, change all variables to be lower case, except for class names

Reply:
These are for #19.